graphql_compiler/tests/snapshot_tests/test_cost_estimation_analysis.py

```diff
@@ -412,15 +412,51 @@ def test_distinct_result_set_estimates_with_revisit_counts(self):
             statistics=statistics,
             pagination_keys=pagination_keys,
             uuid4_field_info=uuid4_field_info,
+            edge_constraints={
+                "Animal_ParentOf": EdgeConstraint.AtMostOneSource,
+            },
         )
 
+        # The Animal_ParentOf edge has an AtMostOneSource constraint, meaning that each
+        # destination is connected to at most one source.
+        # The query below has a unique filter on the root, so the distinct result set
+        # estimate at the root is 1. However, since that animal has at most one parent,
+        # the distinct result set estimate at that location is 1 as well, even though
+        # there are no explicit filters on it.
         query = QueryStringWithParameters(
             """{
             Animal {
                 name @filter(op_name: "=", value: ["$animal_name"])
                 in_Animal_ParentOf @optional {
-                    name @output(out_name: "child_name")
+                    name @output(out_name: "parent_name")
```

Comment:
You can see that the test was using `in_Animal_ParentOf`, which is the opposite of `ParentOf`, meaning `Parent`, but calling the destination a child. That was the source of an incorrect test. I recommend looking at the raw file while reviewing, github didn't do a good job diffing.